iroh/src/magicsock/remote_map/remote_state.rs

```diff
@@ -856,34 +865,58 @@ impl RemoteStateActor {
             .filter_map(|(addr, rtts)| rtts.into_iter().min().map(|rtt| (addr, rtt)))
             .collect();
 
-        // Find the fastest direct or relay path.
-        const IPV6_RTT_ADVANTAGE: Duration = Duration::from_millis(3);
-        let direct_path = path_rtts
+        // Find the fastest direct IPv4 path.
+        let direct_path_ipv4 = path_rtts
             .iter()
-            .filter(|(addr, _rtt)| addr.is_ip())
-            .map(|(addr, rtt)| {
-                if addr.is_ipv4() {
-                    (*rtt + IPV6_RTT_ADVANTAGE, addr)
+            .filter_map(|(addr, rtt)| {
+                if let transports::Addr::Ip(SocketAddr::V4(addr)) = *addr {
+                    Some((addr, *rtt))
                 } else {
-                    (*rtt, addr)
+                    None
                 }
             })
-            .min();
-        let selected_path = direct_path.or_else(|| {
-            // Find the fasted relay path.
-            path_rtts
-                .iter()
-                .filter(|(addr, _rtt)| addr.is_relay())
-                .map(|(addr, rtt)| (*rtt, addr))
-                .min()
+            .min_by_key(|(_addr, rtt)| *rtt);
+
+        // Find the fastest direct IPv6 path.
+        let direct_path_ipv6 = path_rtts
+            .iter()
+            .filter_map(|(addr, rtt)| {
+                if let transports::Addr::Ip(SocketAddr::V6(addr)) = *addr {
+                    Some((addr, *rtt))
+                } else {
+                    None
+                }
+            })
+            .min_by_key(|(_addr, rtt)| *rtt);
+
+        // Find the fastests relay path.
+        let relay_path = path_rtts
+            .iter()
+            .filter(|(addr, _rtt)| addr.is_relay())
+            .min_by_key(|(_addr, rtt)| *rtt)
+            .map(|(addr, rtt)| (addr.clone(), *rtt));
+
+        let current_path = self.selected_path.get().map(|addr| {
+            let rtt = path_rtts.get(&addr).copied();
+            (addr, rtt)
```

Comment:
I'm not sure about this. If the currently selected path is not available in `path_rtts` then it is not an option to keep it. It **must** not be selected because it no longer exists.

So the type of `current_path` should be `Option<(transport:Addr, Duration)>`. This has an impact on `select_best_path` as well.

Reply:
done